sssd_test_framework/utils/sssd.py

```diff
@@ -783,3 +783,50 @@ def autofs(self) -> None:
         """
         self.sssd.authselect.select("sssd")
         self.sssd.enable_responder("autofs")
+
+    def proxy(self,
+              proxy: str,
+              provider: str | list[str] = 'id',
+              role: MultihostRole = None,
+              domain: str | None = None,):
+        """
+        :param domain: Existing domain name, defaults to None (= default domain)
+        :type domain: str | None, optional
+        :param proxy: ``ldap`` or ``files``
+        :type proxy: str
+        :param provider: SSSD providers(id, auth, chpass, ...)
+        :type provider: SSSD providers
+        :param role: MultihostRole(LDAP server)
+        :type role: MultihostRole
+        """
+        target = "sssdproxyldap"
+        if domain is None:
+            domain = self.sssd.default_domain
+
+        match proxy:
+            case "ldap":
+                self.sssd.fs.write('/etc/pam.d/sssdproxyldap', '''
+                        auth    required pam_ldap.so
+                        account required pam_ldap.so
+                        password required pam_ldap.so
+                        session required pam_ldap.so
+                        ''')
+                self.sssd.fs.write('/etc/nslcd.conf', f'uid nslcd\ngid ldap\nuri '
+                                                 f'ldap://{role.host.hostname}\n', dedent=False)
+                self.sssd.svc.restart("nslcd")
+            case "files":
+                target = "sssdproxyfiles"
+                self.sssd.fs.write('/etc/pam.d/sssdproxyfiles', '''
+                        auth       required     pam_unix.so nullok_secure
+                        auth       required     pam_setcred.so
+                        session    required     pam_unix.so 
+                        password   required     pam_unix.so nullok obscure minlen=8
+                        ''')
+            case _:
+                raise Exception("Proxy can be ldap or files but nothing was provided .")
+
+        self.sssd.config[f'domain/{domain}'] = {
+            f'{provider}' + '_provider': 'proxy',
```

Comment:
You don't handle the case where provider is list.

```python
options = {
    "proxy_lib_name": proxy,
    "proxy_pam_target": target,
}

for x in to_list(provider):
    options[f"{x}_provider"] = "proxy"

self.sssd.dom(domain).update(options)
```

Reply:
If we update :

[domain/test]
ldap_tls_reqcert = demand
ldap_tls_cacert = /data/certs/ca.crt
dns_discovery_domain = ldap.test
id_provider = proxy
ldap_uri = ldap://master.ldap.test
proxy_lib_name = ldap
proxy_pam_target = sssdproxyldap
proxy_fast_alias = true
debug_level = 0xfff0

We dont need all the parameters but only 

            "proxy_lib_name": proxy,
            "proxy_pam_target": target,
           [f"{x}_provider"] = "proxy"

Reply:
Both Works

Reply:
But there is also test where you have id_provider=ldap and chpass_provider=proxy. In that case you need to keep the existing stuff. So maybe add a switch to decide if it should override or update the configuration?

Reply:
Update config works . So better to keep it that way . Its was just a info .